src/engine/enginebuffer.h

```diff
@@ -346,11 +347,12 @@ class EngineBuffer : public EngineObject {
 
     // Object used to perform waveform scaling (sample rate conversion)
     EngineBufferScale* m_pScale;
+    FRIEND_TEST(EngineBufferTest, SlowRubberBand);
     // Object used for linear interpolation scaling of the audio
-    EngineBufferScaleLinear* m_pScaleLinear;
+    EngineBufferScale* m_pScaleLinear;
     // Object used for pitch-indep time stretch (key lock) scaling of the audio
-    EngineBufferScaleST* m_pScaleST;
-    EngineBufferScaleRubberBand* m_pScaleRB;
+    EngineBufferScale* m_pScaleST;
+    EngineBufferScale* m_pScaleRB;
```

Comment:
It looks like I can reduce this a bit, but I still need to make the linear scaler a base class pointer so the injection works.